crengine/src/private/lvfreetypeface.cpp

```diff
@@ -1506,20 +1524,21 @@ lUInt16 LVFreeTypeFace::measureText(const lChar32 *text,
                 flags[t] = GET_CHAR_FLAGS(text[t]);
                 // It seems each soft-hyphen is in its own cluster, of length 1 and width 0,
                 // so HarfBuzz must already deal correctly with soft-hyphens.
-                if (cur_width == prev_width) {
+                if (cur_width_26_6 == prev_width_26_6) {
                     // But if there is no advance (this happens with soft-hyphens),
                     // flag it and don't add any letter spacing.
                     flags[t] |= LCHAR_IS_CLUSTER_TAIL;
                 }
                 else {
-                    cur_width += letter_spacing; // only between clusters/graphemes
-                }            
+                    cur_width_26_6 += letter_spacing_26_6; // only between clusters/graphemes
+                }
             }
+            int cur_width = FONT_METRIC_TO_PX(cur_width_26_6);
             widths[t] = cur_width;
             #ifdef DEBUG_MEASURE_TEXT
                 printf("=> %d (flags=%d) => W=%d\n", cur_width - prev_width, flags[t], cur_width);
             #endif
```

Comment:
Oh, my not updated DEBUG_MEASURE_TEXT statements that won't compile ! :)
But please don't fix them (or really at the very end of your work, or let that to me, I'm not yet sure what I will want to see there, px or 26_6 units).

Reply:
Ok